test/ApplicationTest.php

```diff
@@ -240,6 +241,55 @@ public function testRouteAndDispatchMiddlewareAreNotPipedAtInstantation()
         $this->assertCount(0, $pipeline);
     }
 
+    public function testDispatchMiddlewareCanDispatchArrayOfMiddlewareAsMiddlewarePipe()
+    {
+        $middleware = [
+            function () {
+            },
+            'FooBar',
+            [InvokableMiddleware::class, 'staticallyCallableMiddleware'],
+            InvokableMiddleware::class,
+        ];
+
+        $middlewareRoute = function ($req, $res, $next) {
+        };
+        $request = new Request([], [], '/', 'GET');
+        $routeResult = RouteResult::fromRoute(new Route('/', $middlewareRoute), $middleware);
+        $request = $request->withAttribute(RouteResult::class, $routeResult);
+
+        $container = $this->mockContainerInterface();
+        $this->injectServiceInContainer($container, 'FooBar', function () {
+        });
+
+        $app = new Application($this->router->reveal(), $container->reveal());
+        $app->pipeDispatchMiddleware($request, new Response(), function () {
+        });
+    }
+
+
+    public function uncallableMiddleware()
+    {
+        return [
+            ['foo'],
+            [['foo']]
+        ];
+    }
+
+    /**
+     * @dataProvider uncallableMiddleware
+     * @expectException Zend\Expressive\Exception\InvalidMiddlewareException
```

Comment:
Please use `->expectException()` right before the `->pipeDispatchMiddleware()` call